fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/fluid/FluidVariantAttributes.java

```diff
@@ -0,0 +1,159 @@
+
+	/**
+	 * Register an attribute handler for the passed fluid.
+	 */
+	public static void register(Fluid fluid, FluidVariantAttributeHandler handler) {
+		if (HANDLERS.putIfAbsent(fluid, handler) != null) {
+			throw new IllegalArgumentException("Duplicate handler registration for fluid " + fluid);
+		}
+	}
+
+	/**
+	 * Return the attribute handler for the passed fluid, if available, and {@code null} otherwise.
+	 */
+	@Nullable
+	public static FluidVariantAttributeHandler getHandler(Fluid fluid) {
+		return HANDLERS.get(fluid);
+	}
+
+	/**
+	 * Return the attribute handler for the passed fluid, if available, or the default instance otherwise.
+	 */
+	public static FluidVariantAttributeHandler getHandlerOrDefault(Fluid fluid) {
+		FluidVariantAttributeHandler handler = HANDLERS.get(fluid);
+		return handler == null ? DEFAULT_HANDLER : handler;
+	}
+
+	/**
+	 * Return the name that should be used for the passed fluid variant.
+	 */
+	public static Text getName(FluidVariant variant) {
+		return getHandlerOrDefault(variant.getFluid()).getName(variant);
+	}
+
+	/**
+	 * Return the sound corresponding to a container of this fluid variant being filled.
+	 */
+	public static SoundEvent getFillSound(FluidVariant variant) {
+		return getHandlerOrDefault(variant.getFluid()).getFillSound(variant);
+	}
+
+	/**
+	 * Return the sound corresponding to a container of this fluid variant being emptied.
+	 */
+	public static SoundEvent getEmptySound(FluidVariant variant) {
+		return getHandlerOrDefault(variant.getFluid()).getEmptySound(variant);
+	}
+
+	/**
+	 * Return an integer in [0, 15]: the light level emitted by this fluid variant, or 0 if it doesn't naturally emit light.
+	 */
+	public static int getLuminance(FluidVariant variant) {
+		int luminance = getHandlerOrDefault(variant.getFluid()).getLuminance(variant);
+
+		if (luminance < 0 || luminance > 15) {
+			TransferApiImpl.LOGGER.warn("Broken FluidVariantAttributeHandler. Invalid luminance %d for fluid variant %s".formatted(luminance, variant));
+			return DEFAULT_HANDLER.getLuminance(variant);
+		}
+
+		return luminance;
+	}
+
+	/**
+	 * Return a non-negative integer, representing the temperature of this fluid in Kelvin.
+	 * The reference values are 300 for water, and 1300 for lava.
+	 */
+	public static int getTemperature(FluidVariant variant) {
+		int temperature = getHandlerOrDefault(variant.getFluid()).getTemperature(variant);
+
+		if (temperature < 0) {
+			TransferApiImpl.LOGGER.warn("Broken FluidVariantAttributeHandler. Invalid temperature %d for fluid variant %s".formatted(temperature, variant));
+			return DEFAULT_HANDLER.getTemperature(variant);
+		}
+
+		return temperature;
+	}
+
+	/**
+	 * Return a positive integer, representing the viscosity of this fluid variant.
+	 * Fluids with lower viscosity generally flow faster than fluids with higher viscosity.
+	 *
+	 * <p>More precisely, viscosity should roughly be 200 * {@code FlowableFluid.getFlowSpeed}.
```

Comment:
```suggestion
	 * <p>More precisely, viscosity should be {@value FluidConstants#VISCOSITY_RATIO} *
	 * {@link net.minecraft.fluid.FlowableFluid#getFlowSpeed()} for flowable fluids.
```
The `@value` tag automatically fills in the constant value